sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala

```diff
@@ -115,40 +111,8 @@ object RequireSingleBatch extends CoalesceGoal {
 
 case class TargetSize(override val targetSizeBytes: Long) extends CoalesceGoal
 
-class RemoveEmptyBatchIterator(iter: Iterator[ColumnarBatch],
```

Comment:
This was only used in `AbstractGpuCoalesceIterator` but it also had its own `onDeck` that I wanted to get rid of so the functionality was merged into the iterator.